controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -13,6 +13,15 @@ const InitDiscoveryTopicData = async (req, res) => {
   const userId = req.userId;
 
   try {
+    let totalDataQuery = `SELECT 
+                              count(C.topic_id) as total_data
+                          FROM user_topics A 
+                          INNER JOIN user_topics B 
+                              ON A.topic_id = B.topic_id 
+                              AND A.user_id = :userId
+                          RIGHT JOIN topics C 
+                              ON C.topic_id = A.topic_id`;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query for `totalDataQuery` seems to be incorrect. It's joining `user_topics` with itself which doesn't seem necessary. Also, the `RIGHT JOIN` with `topics` table might not give the correct count if there are topics that no user has yet subscribed to. Consider revising the query to directly count the number of topics a user is subscribed to.
```diff
-    let totalDataQuery = `SELECT 
-                              count(C.topic_id) as total_data
-                          FROM user_topics A 
-                          INNER JOIN user_topics B 
-                              ON A.topic_id = B.topic_id 
-                              AND A.user_id = :userId
-                          RIGHT JOIN topics C 
-                              ON C.topic_id = A.topic_id`;
+    let totalDataQuery = `SELECT 
+                              COUNT(topic_id) as total_data
+                          FROM user_topics
+                          WHERE user_id = :userId`;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->